js/web/lib/wasm/jsep/webgpu/ops/pad.ts

```diff
@@ -35,171 +34,167 @@ const validateInputs = (inputs: readonly TensorView[]): void => {
                 }
-                if (k >= ${inputDims[i]}) {
-                  k -= ${inputDims[i]};
+                if (k >= i32(${getElementAt('uniforms.x_shape', i, inputRank)})) {
+                  k -= i32(${getElementAt('uniforms.x_shape', i, inputRank)});
                 }
-                offset += k * ${inputStrides[i]};
+                offset += k * i32(${getElementAt('uniforms.x_strides', i, inputRank)});
             `;
-      }
+  }
 
-      return `
+  return `
               var offset = 0;
               var k = 0;
               ${block}
               value = x[offset];
           `;
-    };
+};
 
 const getPadSnippet =
-    (output: IndicesHelper, inputDims: readonly number[], inputStrides: readonly number[], attributes: PadAttributes,
-     dataType: string): string => {
+    (output: IndicesHelper, inputRank: number, attributes: PadAttributes, dataType: string): string => {
       switch (attributes.mode) {
         case 0:
-          return getPadConstant(output, inputDims, inputStrides, attributes.pads, dataType, attributes.value);
+          return getPadConstant(output, inputRank, attributes.pads.length, dataType);
         case 1:
-          return getPadReflect(output, inputDims, inputStrides, attributes.pads);
+          return getPadReflect(output, inputRank, attributes.pads.length);
         case 2:
-          return getPadEdge(output, inputDims, inputStrides, attributes.pads);
+          return getPadEdge(output, inputRank, attributes.pads.length);
         case 3:
-          return getPadWrap(output, inputDims, inputStrides, attributes.pads);
+          return getPadWrap(output, inputRank, attributes.pads.length);
         default:
           throw new Error('Invalid mode');
       }
     };
 
-const generatePadCode =
-    (shaderHelper: ShaderHelper, inputs: readonly TensorView[], attributes: PadAttributes, dataType: string):
-        string => {
-          const inputDims = inputs[0].dims;
-          const outputDims = ShapeUtil.padShape(inputDims.slice(), attributes.pads);
-          const outputSize = ShapeUtil.size(outputDims);
-          const inputStrides = ShapeUtil.computeStrides(inputDims);
-
-          const output = outputVariable('output', inputs[0].dataType, outputDims);
-          const input = inputVariable('x', inputs[0].dataType, inputDims);
-
-          const padSnippet = getPadSnippet(output, inputDims, inputStrides, attributes, dataType);
-          const padCode = `
-              ${shaderHelper.declareVariables(input, output)}
-              ${shaderHelper.mainStart()}
-              ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes(outputSize)}
+const createPadProgramInfo = (inputs: readonly TensorView[], attributes: PadAttributes): ProgramInfo => {
+  const outputShape = ShapeUtil.padShape(inputs[0].dims.slice(), attributes.pads);
+  const inputDims = inputs[0].dims;
+  const outputDims = ShapeUtil.padShape(inputDims.slice(), attributes.pads);
```

Comment:
It seems that `outputDims` is same with `outputShape`. Please unify to use one.

Reply:
done